src/LtiMessageLaunch.php

```diff
@@ -122,6 +122,49 @@ public static function fromCache(
         return $new->validateRegistration();
     }
 
+    public function initialize(array $request = null): static
+    {
+        $this->setRequest($request);
+
+        // Ideally this would just be abstracted to a validator. But how when, halfway through, we need to migrate?
+        $this->validateState()
+            ->validateJwtFormat()
+            ->validateNonce()
+            ->validateRegistration()
+            ->validateJwtSignature();
+
+        // This feels kinda icky in the middle of all the validation. How can I move it?
+        if ($this->shouldMigrate()) {
+            $this->deployment = $this->db->migrateFromLti11($this->getLaunchData());
+        }
+
+        $this->validateDeployment()
+            ->validateMessage();
+
+        $this->cacheLaunchData();
+
+        return $this;
+    }
+
+    public function setRequest(array $request = null)
+    {
+        if ($request === null) {
+            $request = $_POST;
+        }
+
+        $this->request = $request;
+
+        return $this;
+    }
+
+    private function shouldMigrate(): bool
+    {
+        return $this->hasMigrationStrategy()
```

Comment:
Where is `hasMigrationStrategy()` defined?

Reply:
It's not and I wasn't sure where it should be. It depends on how we end up defining the migration process.